### ISSUE TITLE
Add DiscoveryMode to DiscoveryStrategy

### ISSUE THREAD

**Issue by gokhanoner** [hazelcast/hazelcast#14321]
`DiscoveryService` know the `DiscoveryMode`, `Member` or `Client` but don't use this when creating `DiscoveryStrategy` object. This could be usefull when there are some logic changes between member & client discovery, especially in Cloud discoveries like defining extra checks & fail-fast when using a client without some additional properties etc. Currently, the only way of doing this to either warn users about those properties or add a discovery property for it and ask the user to define it.

**Comment by gokhanoner** [comment]
@mmedenjak one option could be adding this to `properties` that passed to `DiscoveryStrategy` constructor. This way we don't need to add another method to a public interface & even do the change in a patch release?

**Comment by mmedenjak** [comment]
@gokhanoner yes, it's possible. I don't see this as critical for now so moving it.
